### PR TITLE
M2 trip details Date & Test

### PR DESCRIPTION
**Pull request by kanatagu**
# Description
* Changed Format for date
  * If you use Date object, the ISO time convert into the user's OS time zone so I just used it to calculate the order or such things, not to show.
* If activities of timeFrom or timeTo across the multiple dates, the start date or end date is showed like (01/03). You can see with screen shot

* Added Integration test for Trip Details Tabs as it's complicated part

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Check List
* Date format works
* Test Pass

For test, you can run with this command
```
pnpm test

or 

pnpm test -t TripDetailsTabs
```

## Screen shot
Screenshot 2023-12-24 at 7 55 03 PM

Screenshot 2023-12-24 at 7 55 14 PM

### CODE REVIEW

File: app/trip/[id]/components/trip-details-tabs.tsx [hunk 1]
```diff
@@ -28,51 +28,71 @@ type TripDetailsTabsProps = {
 export const TripDetailsTabs = ({ activities }: TripDetailsTabsProps) => {
   const borderColor = useColorModeValue('gray.300', 'gray.600')
 
-  // Sorting by date Time
-  const sortedActivities = activities.sort((a, b) => {
-    const dateA = new Date(a.timeFrom).getTime()
-    const dateB = new Date(b.timeFrom).getTime()
+  const sortedActivities = useMemo(() => {
```
Comment (samuraikun) [inline app/trip/[id]/components/trip-details-tabs.tsx:31]: Nice use of useMemo to optimize performance by avoiding unnecessary recalculations! :+1:

**Comment by samuraikun** [comment]
@Kana-TAGUCHI Almost good! 👍 
If possible, could we consider adding a test CI job on GitHub Actions like this?

ex. `.github/workflows/test.yml`

```yaml
name: Run Jest Tests

on: [push]

jobs:
  test:
    name: Run tests
    runs-on: ubuntu-latest

    steps:
      - name: Check out Git repository
        uses: actions/checkout@v4

      - name: Setup Node
        uses: actions/setup-node@v4
        with:
          node-version: '20.8.0'

      - name: Install pnpm
        uses: pnpm/action-setup@v2
        with:
          version: 8

      - name: Install Node.js dependencies
        run: |
          pnpm i --frozen-lockfile

      - name: Run Jest Tests
        run: |
          pnpm test

```

**Review by YoshinariYamanaka — APPROVED** [review]

**Comment by kanatagu** [comment]
@samuraikun 
Thank you! I added the github action 🚀 
https://github.com/tabi-memo/frontend/pull/19/commits/8a175b4b1bc8b7ae18610e9bdc6cd5f9e70fedb6

**Review by samuraikun — APPROVED** [review]
🍎

**Pull request merged**